ci-operator/step-registry/gather/gcp-console/gather-gcp-console-commands.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+export GOOGLE_CLOUD_KEYFILE_JSON="${CLUSTER_PROFILE_DIR}/gce.json"
+gcloud auth activate-service-account --key-file="${GOOGLE_CLOUD_KEYFILE_JSON}"
+
+if test ! -f "${SHARED_DIR}/metadata.json"
+then
+	echo "No metadata.json, so unknown GCP project, so unable to gathering console logs."
+	exit 0
+fi
+
+gcloud config set project "$(jq -r .gcp.projectID "${SHARED_DIR}/metadata.json")"
+
+if test -f "${KUBECONFIG}"
+then
+	oc --request-timeout=5s get nodes -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' > "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+
+	oc --request-timeout=5s -n openshift-machine-api get machines -o jsonpath --template '{range .items[*]}{.spec.providerID}{"\n"}{end}' | sed 's|.*/||' >> "${TMPDIR}/node-provider-IDs.txt" &
+	wait "$!"
+else
+	echo "No kubeconfig; skipping providerID extraction."
+fi
+
+if test -f "${SHARED_DIR}/gcp-instance-ids.txt"
+then
+	cat "${SHARED_DIR}/gcp-instance-ids.txt" >> "${TMPDIR}/node-provider-IDs.txt"
+fi
+
+cat "${TMPDIR}/node-provider-IDs.txt" | sort | uniq | while read -r INSTANCE_ID
+do
+	echo "Finding the zone for ${INSTANCE_ID}"
+	ZONE="$(gcloud --format json compute instances list "--filter=name=(${INSTANCE_ID})" | jq -r '.[].zone')" &
+	wait "$!"
```

Comment:
Hrm, I don't understand [this][1] (from [this job][2]):

```
Finding the zone for ci-op-yjsrfl7s-d1016-tpmpm-master-0
/bin/bash: line 44: ZONE: unbound variable
error: failed to execute wrapped command: exit status 1
```

Because I'm trying to set that variable a few lines up:

```console
$ git grep -2hn ZONE ci-operator/step-registry/gather/gcp-console/gather-gcp-console-commands.sh | head -n8
37-do
38-     echo "Finding the zone for ${INSTANCE_ID}"
39:     ZONE="$(gcloud --format json compute instances list "--filter=name=(${INSTANCE_ID})" | jq -r '.[].zone')" &
40-     wait "$!"
41:     if test -z "${ZONE}"
42-     then
43-             echo "No zone found for ${INSTANCE_ID}, so not attempting to gather console logs"
44-     fi
```

Maybe I can't background the variable-set.

[1]: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/13398/rehearse-13398-pull-ci-openshift-cluster-etcd-operator-release-4.6-e2e-agnostic/1339343972186722304/artifacts/e2e-agnostic/gather-gcp-console/container-logs/test.log
[2]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/13398/rehearse-13398-pull-ci-openshift-cluster-etcd-operator-release-4.6-e2e-agnostic/1339343972186722304

Reply:
I've pushed d4bb001b65 -> 5d9d973fbe to move the `wait` inside the subshell.